src/Middleware/HttpLogging/src/HttpRequestLog.cs

```diff
@@ -8,12 +8,12 @@ namespace Microsoft.AspNetCore.HttpLogging;
 
 internal sealed class HttpRequestLog : IReadOnlyList<KeyValuePair<string, object?>>
 {
-    private readonly List<KeyValuePair<string, object?>> _keyValues;
+    private readonly IList<KeyValuePair<string, object?>> _keyValues;
     private string? _cachedToString;
 
     internal static readonly Func<object, Exception?, string> Callback = (state, exception) => ((HttpRequestLog)state).ToString();
 
-    public HttpRequestLog(List<KeyValuePair<string, object?>> keyValues)
+    public HttpRequestLog(IList<KeyValuePair<string, object?>> keyValues)
```

Comment:
I remember @geeknoid mentioning that virtual calls are costly, here's a small benchmark that shows that moving from `List<T>` -> `IList<T>` is 12% worse that it's now: https://github.com/xakep139/IListVsListBench/tree/master/IListVsListBench

Reply:
Does the CA1859 analyzer diagnostic enabled on this repo? It flags a bunch of cases where code can be upgraded to use concrete types in order to eliminate virtual or interface dispatch.

Reply:
Unlikely. In this case the collection is exposed publicly, so we want an abstraction there, but we can pass the concrete type here.